### DISCUSSION TITLE
Debug mode for scripts

### DISCUSSION THREAD

**Discussion by cypof** [nushell/nushell#4530] [Q&A]
What would be the best way to run nu scripts in debug mode? E.g. bash can run scripts with settings like this:
```
# stop and show the stack on any error
set -eE -o functrace
# error on unset variables
set -u
# print each command before executing it
set -x
# extend "set -x" above with source file, line number, function name etc
export PS4='+(${BASH_SOURCE:-""}:${LINENO}) ${FUNCNAME[0]:+${FUNCNAME[0]}(): }'
```

**Comment by kubouch** [comment]
We don't have debugging features like this yet. We'd like to expand more on debugging but so far we have only `debug` and `describe` commands.

**Comment by cypof** [comment]
OK unfortunate, I will create an issue for this then. For info you can get basic logs by adding a couple print statements in `nushell/crates/nu-engine/src/eval.rs`. E.g. at the beginning of `eval_expression_with_input`.

**Reply by GrabbenD** [reply]
For folks coming from Google search, here's the issue: https://github.com/nushell/nushell/issues/4539

Similar ticket: https://github.com/nushell/nushell/issues/10037
